src/hooks/useResolveEns.ts

```diff
@@ -1,31 +1,57 @@
 // React + Web3 Essentials
 import { ethers } from 'ethers';
-import { useContext,useEffect, useState } from 'react';
+import { useWeb3React } from '@web3-react/core';
+import { useContext, useEffect, useState } from 'react';
 
 // Internal Components
 import { caip10ToWallet } from 'helpers/w2w';
 import { AppContext, MessageIPFS } from 'types/chat';
 import { Context } from 'modules/chat/ChatModule';
 import { intitializeDb } from 'components/chat/w2wChat/w2wIndexeddb';
 
-
 // Internal Configs
 import { appConfig } from 'config';
 
-const getEnsFromIndexDb = async(checksumWallet:string):Promise<any> => {
+const getEnsFromIndexDb = async (checksumWallet: string): Promise<any> => {
   const ensFromIndexDB: any = await intitializeDb<string>('Read', 'Wallets', checksumWallet, '', 'ens');
   return ensFromIndexDB;
-}
+};
+
+const getEnsName = async (provider, checksumWallet) => {
+  let ensName: string = '';
+  provider.lookupAddress(checksumWallet).then(async (ens) => {
+    if (ens) {
+      ensName = ens;
+      await intitializeDb<MessageIPFS>('Insert', 'Wallets', checksumWallet, ens, 'ens');
+    } else {
+      ensName = null;
+      await intitializeDb<MessageIPFS>('Insert', 'Wallets', checksumWallet, null, 'ens');
+    }
+  });
+  return ensName;
+};
+
+// const getAddress = async (provider: ethers.providers.BaseProvider | any, ensName: string) => {
+//   let address: string;
+//   let checksummedAddress: string;
+//   address = await provider.resolveName(ensName);
+
+//   // this ensures address are checksummed
+//   checksummedAddress = ethers.utils.getAddress(address.toLowerCase());
+
+//   return checksummedAddress;
+// };
 
 export function useResolveEns(username: string): string {
```

Comment:
Can you try with two optional input params, one as address and one as ens, then we can do the jobs for each, as pe which input is provided?